### PR TITLE
fix: add idempotent in func SFS CreatVolume

### PR DESCRIPTION
**Pull request by deer-hang**

### CODE REVIEW

File: pkg/sfs/controllerserver.go [hunk 1]
```diff
@@ -33,55 +37,74 @@ type controllerServer struct {
...
-    if err != nil {
-		klog.V(3).Infof("Failed to create SFS v2 client: %v", err)
-		return nil, status.Error(codes.Internal, err.Error())
-    }
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create SFS v2 client: %v", err)
+	}
+	name := req.GetName()
+	capacityRange := req.GetCapacityRange()
+	if err := createVolumeValidation(client, name, capacityRange); err != nil {
```
Comment (chengxiangdong) [inline pkg/sfs/controllerserver.go:49]: if name and size are the same, it should return success

File: pkg/sfs/share.go [hunk 2]
```diff
@@ -19,27 +19,29 @@ package sfs
...
 	share, err := shares.Create(client, createOpts).Extract()
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "Failed to create share, %v", err)
 	}
 
 	err = waitForShareStatus(client, share.ID, shareAvailable, waitForAvailableShareTimeout)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "Failed to wait for share create, %v", err)
```
Comment (chengxiangdong) [inline pkg/sfs/share.go:44]: It is recommended to print `Error waiting for share to be created: %v`

File: pkg/sfs/controllerserver.go (hunk 1 above)
Comment (deer-hang) [reply pkg/sfs/controllerserver.go]: done

File: pkg/sfs/share.go (hunk 2 above)
Comment (deer-hang) [reply pkg/sfs/share.go]: done

File: pkg/sfs/controllerserver.go [hunk 3]
```diff
@@ -33,55 +37,88 @@ type controllerServer struct {
...
+		return nil, status.Errorf(codes.Internal,
+			"Failed to query SFS by name, cannot verify whether it exists: %v", err)
+	}
+	for _, v := range list {
+		if v.Name == name {
+			// if name, share proto and size are the same, return success
+			if v.ShareProto == shareProto && v.Size == sizeInGiB {
+				return &v, nil
+			}
+			return nil, status.Errorf(codes.InvalidArgument, "SFS name: %s already exists", name)
```
Comment (chengxiangdong) [inline pkg/sfs/controllerserver.go:116]: suggest to print "SFS name: %s already exists with a different size or share_proto"

Comment (deer-hang) [reply pkg/sfs/controllerserver.go]: done

**Comment by deer-hang** [comment]
Resubmitted PR based on the latest code: 
"[feat-sfs-createvolume](https://github.com/huaweicloud/huaweicloud-csi-driver/pull/32)"

**Pull request closed without merging**